analyzers/src/SonarAnalyzer.CSharp/Rules/ConditionalStructureSameImplementation.cs

```diff
@@ -18,110 +18,140 @@
+            : node.ChildNodes();
+
+        return statements.Count(IsApprovedStatement);
+    }
+
+    private static void ReportSyntaxNode(SonarSyntaxNodeReportingContext context, SyntaxNode node, SyntaxNode precedingNode, string errorMessageDiscriminator) =>
+        context.ReportIssue(Rule, node, [precedingNode.ToSecondaryLocation()], precedingNode.GetLineNumberToReport().ToString(), errorMessageDiscriminator);
+
+    private static bool IsApprovedStatement(SyntaxNode statement) =>
+        !statement.IsAnyKind(IgnoredStatementsInSwitch);
+
+    private static bool HaveTheSameInvocations(SyntaxList<SyntaxNode> first, SyntaxList<SyntaxNode> second, SemanticModel model)
+    {
+        var referenceInvocations = first.SelectMany(x => x.DescendantNodes().OfType<InvocationExpressionSyntax>()).ToArray();
+        var candidateInvocations = second.SelectMany(x => x.DescendantNodes().OfType<InvocationExpressionSyntax>()).ToArray();
+        return HaveTheSameInvocations(referenceInvocations, candidateInvocations, model);
+    }
+
+    private static bool HaveTheSameInvocations(SyntaxNode first, SyntaxNode second, SemanticModel model)
+    {
+        var referenceInvocations = first.DescendantNodes().OfType<InvocationExpressionSyntax>().ToArray();
+        var candidateInvocations = second.DescendantNodes().OfType<InvocationExpressionSyntax>().ToArray();
+        return HaveTheSameInvocations(referenceInvocations, candidateInvocations, model);
+    }
 
-        public static bool HaveTheSameInvocations(SyntaxList<SyntaxNode> first, SyntaxList<SyntaxNode> second, SemanticModel model)
+    private static bool HaveTheSameInvocations(InvocationExpressionSyntax[] referenceInvocations, InvocationExpressionSyntax[] candidateInvocations, SemanticModel model)
+    {
+        if (referenceInvocations.Length != candidateInvocations.Length)
         {
-            var referenceInvocations = first.SelectMany(x => x.DescendantNodes().OfType<InvocationExpressionSyntax>()).ToArray();
-            var candidateInvocations = second.SelectMany(x => x.DescendantNodes().OfType<InvocationExpressionSyntax>()).ToArray();
-            if (referenceInvocations.Length != candidateInvocations.Length)
-            {
-                return false;
-            }
+            return false;
+        }
 
-            for (var i = 0; i < referenceInvocations.Length; i++)
+        for (var i = 0; i < referenceInvocations.Length; i++)
+        {
+            if (!referenceInvocations[i].IsEqualTo(candidateInvocations[i], model))
             {
-                if (!referenceInvocations[i].IsEqualTo(candidateInvocations[i], model))
-                {
-                    return false;
-                }
+                return false;
             }
-            return true;
         }
+        return true;
     }
+
+    private sealed record EquivalentNodeCompare(SyntaxNode RefNode, SyntaxNode OtherNode);
```

Comment:
I used `sealed` because S3260 suggested it.
Should it suggest `readonly record struct` instead?
Is it more performant to use the later rather than `sealed record`?

Reply:
A record is a class and allocates. These kinds of helper types are often used to return multiple values from a method. As such they are comparable to method parameters. These you want to pass allocation free per stack. A struct allows you do do just that.